src/input/l1/testing.rs

```diff
@@ -787,216 +834,330 @@ struct ValidatorInfo {
-                        let schnorr_key =
-                            StateKeyPair::generate_from_seed_indexed(seed, validator_index);
-
-                        let provider = ProviderBuilder::new()
-                            .wallet(EthereumWallet::from(build_signer(DEV_MNEMONIC, 0)))
-                            .connect_http(rpc_url.clone());
-
-                        match StakingTransactions::create(
-                            rpc_url.clone(),
-                            &provider,
-                            stake_table_addr,
-                            vec![(signer.clone(), bls_key.clone(), schnorr_key.clone())],
-                            DelegationConfig::MultipleDelegators,
-                        )
-                        .await
-                        {
-                            Ok(mut staking_txns) => {
-                                let validator_provider =
-                                    staking_txns.provider(signer.address()).unwrap().clone();
-
-                                let delegator_infos: Vec<DelegatorInfo> = staking_txns
-                                    .delegations()
-                                    .iter()
-                                    .filter(|d| d.validator == signer.address())
-                                    .filter_map(|d| {
-                                        let provider = staking_txns.provider(d.from)?.clone();
-
-                                        Some(DelegatorInfo {
-                                            address: d.from,
-                                            provider,
-                                        })
-                                    })
-                                    .collect();
-
-                                if staking_txns.apply_all().await.is_ok() {
-                                    println!(
-                                        "Background: Registered validator #{validator_index} with {} delegators",
-                                        delegator_infos.len(),
-                                    );
-
-                                    registered_validators.push(ValidatorInfo {
-                                        index: validator_index,
-                                        address: signer.address(),
-                                        provider: validator_provider,
-                                        delegators: delegator_infos,
-                                    });
-                                }
-                            }
-                            Err(e) => {
-                                eprintln!(
-                                    "Background: Failed to create staking transactions for validator #{validator_index}: {e:?}"
-                                );
-                            }
-                        }
-                        validator_index += 1;
+struct BackgroundTaskState {
```

Comment:
What about just reusing `EventGenerator` here, and using the generated events to decide which transactions to submit? Since that already generates valid events and keeps track of all this state as well.

Reply:
I was thinking about that, but I'll leave it as it is for now and comeback to this later